src/components/paint-editor/paint-editor.jsx

```diff
@@ -62,126 +63,134 @@ class PaintEditorComponent extends React.Component {
-                            <ButtonGroup>
-                                <Button
-                                    className={styles.buttonGroupButton}
-                                    onClick={this.props.onUndo}
-                                >
-                                    <img
-                                        alt="Undo Icon"
-                                        className={styles.buttonGroupButtonIcon}
-                                        src={undoIcon}
-                                    />
-                                </Button>
-                                <Button
-                                    className={styles.buttonGroupButton}
-                                    onClick={this.props.onRedo}
-                                >
-                                    <img
-                                        alt="Redo Icon"
-                                        className={styles.buttonGroupButtonIcon}
-                                        src={redoIcon}
+                {this.state.canvas ? (
+                    <div className={styles.editorContainerTop}>
+                        {/* First row */}
+                        <div className={styles.row}>
+                            {/* Name field */}
+                            <InputGroup>
+                                <Label text={this.props.intl.formatMessage(messages.costume)}>
+                                    <BufferedInput
+                                        type="text"
+                                        value={this.props.name}
+                                        onSubmit={this.props.onUpdateName}
                                     />
-                                </Button>
-                            </ButtonGroup>
-                        </InputGroup>
+                                </Label>
+                            </InputGroup>
 
-                        {/* Group/Ungroup */}
-                        <InputGroup className={styles.modDashedBorder}>
-                            <LabeledIconButton
-                                imgAlt="Group Icon"
-                                imgSrc={groupIcon}
-                                title="Group"
-                                onClick={this.props.onGroup}
-                            />
-                            <LabeledIconButton
-                                imgAlt="Ungroup Icon"
-                                imgSrc={ungroupIcon}
-                                title="Ungroup"
-                                onClick={this.props.onUngroup}
-                            />
-                        </InputGroup>
+                            {/* Undo/Redo */}
+                            <InputGroup>
+                                <ButtonGroup>
+                                    <Button
+                                        className={styles.buttonGroupButton}
+                                        onClick={this.props.onUndo}
```

Comment:
Not necessarily in this PR, but we should add disabled states for undo and redo too. (The functions aren't pre-existing in helper/undo, but if you look in reducers/undo, they would look like those functions that log "Can't undo" and "can't redo")

Reply:
canUndo: return (state.pointer > 1)
canRedo: (state.pointer > -1 && state.pointer !== state.stack.length - 1)